src/info/width.rs

```diff
@@ -0,0 +1,28 @@
+use crate::data::{Digit, DoubleDigit, ApInt};
+use crate::info::{BitWidth};
+
+/// Types that have an associated bit width may implement `Width`.
+pub trait Width {
+    /// Returns the bit width of `self`.
+    fn width(&self) -> BitWidth;
+}
+
+impl Width for Digit {
+    fn width(&self) -> BitWidth {
+        BitWidth::from(Digit::BITS)
+    }
+}
+
+impl Width for DoubleDigit {
+    fn width(&self) -> BitWidth {
+        BitWidth::from(Digit::BITS * 2)
+    }
+}
+
+impl Width for ApInt {
+    /// Returns the `BitWidth` of this `ApInt`.
+    #[inline]
+    fn width(&self) -> BitWidth {
+        BitWidth::new(self.len_bits()).unwrap()
+    }
```

Comment:
Hmm what about `Int` and `Uint`?
Are they defined in their respective module?
Why wasn't that done for `Digit`, `DoubleDigit` and `Apint`? (seems to be a bit inconsistent)

Reply:
I moved them into their respective modules. I am glad you pointed this out, because it looks like `Width` has never been implemented for `Int` or `Uint`.